reconcile/dashdotdb_dora.py

```diff
@@ -0,0 +1,480 @@
+                    saastarget = SaasTarget(
+                        app_name=saas_file.name,
+                        env_name=target_env_name,
+                        path=saas_file_path,
+                        resource_template=rt.name,
+                        namespace=ns.path,
+                        pipeline=pipeline,
+                    )
+
+                    targets.append(saastarget)
+
+                    LOG.info("marking SaaS file for processing: %s", saastarget)
+        return targets
+
+    def get_app_env_deployments(
+        self, app_env: AppEnv
+    ) -> tuple[AppEnv, list[Deployment]]:
+        since = self.get_latest(app_env)
+
+        if not since:
+            # grab last 90 days
+            since = self.start_timestamp - timedelta(days=90)
+
+        return (
+            app_env,
+            self.get_dep_db_deployments(app_env, since),
+        )
+
+    def get_repo_changes(
+        self, saastarget_deployment: tuple[SaasTarget, Deployment]
+    ) -> tuple[SaasTarget, Deployment, RepoChanges]:
+        saastarget, deployment = saastarget_deployment
+        LOG.info("Fetching repo changes for %s - %s", saastarget, deployment)
+        # sometimes the trigger_reason ends with '[auto-promotion]', so we keeep
+        # only the first 40 chars.
+        promotion_sha = deployment.trigger_reason.split("/")[-1][:40]
+
+        repo_url, ref_from = self.get_repo_ref_for_sha(saastarget, promotion_sha + "^")
+        _, ref_to = self.get_repo_ref_for_sha(saastarget, promotion_sha)
+
+        return (saastarget, deployment, RepoChanges(repo_url, ref_from, ref_to))
+
+    def get_repo_ref_for_sha(
+        self, saastarget: SaasTarget, sha: str
+    ) -> tuple[Optional[str], Optional[str]]:
+        saas_file_yaml = self.app_interface_get_file(saastarget.path, ref=sha).decode()
+        saas_file = yaml.safe_load(saas_file_yaml)
+
+        for rt in saas_file["resourceTemplates"]:
+            if saastarget.resource_template != rt["name"]:
+                continue
+
+            for target in rt["targets"]:
+                if saastarget.namespace == target["namespace"]["$ref"]:
+                    return (rt["url"], target["ref"])
+
+        return (None, None)
+
+    def get_dep_db_deployments(
+        self, app_env: AppEnv, since: datetime
+    ) -> list[Deployment]:
+        LOG.info(
+            "Fetching deployments from DeploymentDB for %s since %s",
+            app_env,
+            since,
+        )
+        return self.dep_db.deployments(app_env, since)
+
+    def compare(self, rc: RepoChanges) -> tuple[RepoChanges, list[Commit]]:
+        if not rc.repo_url:
+            return rc, []
+
+        LOG.info("Fetching commits %s", rc)
+        if rc.repo_url.startswith("https://github.com"):
+            commits = self._github_compare_commits(rc)
+        elif rc.repo_url.startswith("https://gitlab.cee.redhat.com"):
+            commits = self._gitlab_compare_commits(rc)
```

Comment:
i'm not entirely sold on mentioning gitlab.cee explicitely here but i have no ideas that are considerably better :(

Reply:
good point, I think it has an easy fix

Reply:
thank you for `rc.repo_url.startswith(self.gl.server)`!